Examples/AddNftAllowance/main.swift

```diff
@@ -0,0 +1,170 @@
+            fatalError("Failed to create NFT")
+        }
+        
+        print("Created NFT with token ID: \(nftTokenId)")
+        
+        // Step 2: Mint NFTs
+        let cids = [
+            "QmNPCiNA3Dsu3K5FxDPMG5Q3fZRwVTg14EXA92uqEeSRXn",
+            "QmZ4dgAgt8owvnULxnKxNe8YqpavtVCXmc1Lt2XajFpJs9",
+            "QmPzY5GxevjyfMUF5vEAjtyRoigzWp47MiKAtLBduLMC1T"
+        ]
+        
+        for cid in cids {
+            let mintReceipt = try await TokenMintTransaction()
+                .tokenId(nftTokenId)
+                .metadata(cid.data(using: .utf8)!)
+                .freezeWith(client)
+                .execute(client)
+                .getReceipt(client)
+            
+            guard let serials = mintReceipt.serials else {
+                fatalError("Failed to mint NFT")
+            }
+            
+            print("Minted NFT (token ID: \(nftTokenId)) with serial: \(serials.first!)")
+        }
+        
+        // Step 3: Create spender and receiver accounts
+        let spenderKey = PrivateKey.generateEd25519()
+        let receiverKey = PrivateKey.generateEd25519()
+        
+        let spenderAccountId = try await AccountCreateTransaction()
+            .key(spenderKey.publicKey)
+            .initialBalance(Hbar(2))
+            .execute(client)
+            .getReceipt(client)
+            .accountId!
+        
+        let receiverAccountId = try await AccountCreateTransaction()
+            .key(receiverKey.publicKey)
+            .initialBalance(Hbar(2))
+            .execute(client)
+            .getReceipt(client)
+            .accountId!
+        
+        print("Created spender account ID: \(spenderAccountId), receiver account ID: \(receiverAccountId)")
+        
+        // Step 4: Associate spender and receiver with the NFT
+        try await TokenAssociateTransaction()
+            .accountId(spenderAccountId)
+            .tokenIds([nftTokenId])
+            .freezeWith(client)
+            .sign(spenderKey)
+            .execute(client)
+            .getReceipt(client)
+        
+        try await TokenAssociateTransaction()
+            .accountId(receiverAccountId)
+            .tokenIds([nftTokenId])
+            .freezeWith(client)
+            .sign(receiverKey)
+            .execute(client)
+            .getReceipt(client)
+        
+        print("Associated spender and receiver accounts with NFT")
+        
+        // Step 5: Approve NFT allowance for spender
+        try await AccountAllowanceApproveTransaction()
+            .approveTokenNftAllowance(NftId(tokenId: nftTokenId, serial: 1), env.operatorAccountId, spenderAccountId)
+            .approveTokenNftAllowance(NftId(tokenId: nftTokenId, serial: 2), env.operatorAccountId, spenderAccountId)
+            .execute(client)
+        
+        print("Approved NFT allowance for spender")
+        
+        // Step 6: Transfer NFT using approved allowance
+        let transferReceipt = try await TransferTransaction()
```

Comment:
`.transactionId(TransactionId.generateFrom(spenderAccountId))` should be added to this in order for the transaction to be executed on behalf of the Spender account.